pkg/kubectl/cmd/auth/cani.go

```diff
@@ -93,8 +105,12 @@ func NewCmdCanI(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.C
 		Example: canIExample,
 		Run: func(cmd *cobra.Command, args []string) {
 			cmdutil.CheckErr(o.Complete(f, args))
-			cmdutil.CheckErr(o.Validate())
+			if o.List {
+				cmdutil.CheckErr(o.RunAccessList())
```

Comment:
There should only one `Run` command, which should pick to run a specific request or list under the cover. Don't break the patterns we've set.